### PR TITLE
FEAT: URL input and share

### PR DESCRIPTION
**Pull request by MelanieH7**
### What's Changed
1. Implemented URL feature for algorithm navigation and parameter input.
2. Implemented share feature for generating URL of the current algorithm page.
3. Included relevant example usage and code implementation guidelines under docs: `URL-Code-Implementation-Guideline-20241004.pdf`.

### Screenshots
Screenshot of using URL to navigate to an algorithm page with input parameters.
Screenshot of using URL to navigate to an algorithm page without input parameters.
Screenshot of generating current page URL using the share button.

### Checklist
- [x] Verified on desktop

### How to Test
1. Clone the repository and run `npm install`.
2. Start the server and frontend with `npm start`.
3. Copy URL of the current algorithm page using the share icon button located on top-left of the mid panel.
4. Paste URL into a new browser to navigate to the corresponding algorithm page with parameters.

### Vercel Deployment
- [Live Preview](https://algorithms-in-action-git-2024sem2-aia-ui.vercel.app)

### CODE REVIEW

**Comment by MaryHuang8** [comment]
@Sraccity Ready for review. Resolved conflicts, updated features guideline and prettified commit message. Let us know any further steps needed! :) Thanks.

File: docs/URL-Code-Implementation-Guideline-20241004.pdf
Comment (Sraccity) [inline docs/URL-Code-Implementation-Guideline-20241004.pdf:1]: Note (future Ref.): Documentation File for implementation

File: src/algorithms/parameters/helpers/urlHelpers.js
Comment (Sraccity) [inline src/algorithms/parameters/helpers/urlHelpers.js:1]: Note (Future ref) : Main Helper file

File: src/context/urlState.js
Comment (Sraccity) [inline src/context/urlState.js:1]: Note (Future Ref) : Another URL file

**Review by Sraccity — APPROVED** [review]
**Praise** : Seems like a lot of work was went into it and congrats on your merge


**Comment**: Ngl, I thought this could be a global change but it seems like you guys might have the better approach here. 

**Suggestion**: Do mention things to improve on your handover documentation

- [x] Tested in a local env.?

**Comment by Sraccity** [comment]
**Kudo:** Congrats on your PR
@MelanieH7 @MaryHuang8 @MinniMinMin

**Comment by MelanieH7** [comment]
> **Kudo:** Congrats on your PR @MelanieH7 @MaryHuang8 @MinniMinMin

🥳🥳🥳
Thanks @Sraccity! 🫡👍

File: src/context/urlState.js
Comment (MaryHuang8) [reply src/context/urlState.js:1]: This is for extracting current parameter inputs for the share feature.

File: src/algorithms/parameters/helpers/urlHelpers.js
Comment (MaryHuang8) [reply src/algorithms/parameters/helpers/urlHelpers.js:1]: This is for parsing parameters in the URL and helping to wrap them to the respective <algorithm>param.js file.

**Pull request merged**